pkg/prometheus/servicemonitor.go

```diff
@@ -2,85 +2,23 @@ package prometheus
 
 import (
 	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
-	monitoringclient "github.com/coreos/prometheus-operator/pkg/client/versioned"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/tools/clientcmd"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
-var smLog = logf.Log.WithName("common_serviceMonitor")
+var smLog = logf.Log.WithName("prometheus_serviceMonitor")
 
-func newMonitoringClient() (*monitoringclient.Clientset, error) {
-	config, err := clientcmd.BuildConfigFromFlags("", "")
-	if err != nil {
-		return nil, err
-	}
-	monitoringClient, err := monitoringclient.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
-	return monitoringClient, err
-}
+const (
+	ServiceMonitorNotFound string = "Service Monitor could not be found"
+)
 
-func CreateOrUpdateServiceMonitors(serviceMonitorName, port, namespace string, labels map[string]string) error {
-	serviceMonitor := createServiceMonitor(serviceMonitorName, port, namespace, labels)
+func ServiceMonitorExists(serviceMonitorName, namespace string) (*monitoringv1.ServiceMonitor, error) {
+	smLog.WithValues("Service Monitor", serviceMonitorName, "Namespace", namespace)
 	serviceMonitorClient, err := newMonitoringClient()
-	oldServiceMonitor, err := serviceMonitorClient.Monitoring().ServiceMonitors(namespace).Get(serviceMonitorName, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		_, err = serviceMonitorClient.Monitoring().ServiceMonitors(namespace).Create(serviceMonitor)
-		if err != nil {
-			smLog.Error(err, "Could not create service monitor", "ServiceMonitor :", serviceMonitor)
-			return err
-		}
-		smLog.Info("Created service monitor")
-		return err
-	}
-	if err != nil {
-		smLog.Error(err, "Could not retrieve service monitor", "ServiceMonitor :", serviceMonitor)
-		return err
-	}
-	serviceMonitor.ResourceVersion = oldServiceMonitor.ResourceVersion
-	_, err = serviceMonitorClient.Monitoring().ServiceMonitors(namespace).Update(serviceMonitor)
+	serviceMonitor, err := serviceMonitorClient.Monitoring().ServiceMonitors(namespace).Get(serviceMonitorName, metav1.GetOptions{})
```

Comment:
Do we run some lint system for code? We should.

Reply:
@shtripat , we will be adding them as part of CI tool system (yet tobe added). Once the go-lints are configured we may have to make some minor modifications to the existing code...

Reply:
I already do it in my editor. Any hard checks will be taken care on CI.